tests/ReturnTypeTest.php

```diff
@@ -699,6 +699,147 @@ function test() {
+                     * @template U
+                     * @param callable(T): U $predicate
+                     * @return Closure(iterable<T>): iterable<U>
+                     */
+                    function map(callable $predicate): callable {
+                        return
+                        /**
+                         * @param iterable<T> $iter
+                         * @return iterable<U>
+                         */
+                        function(iterable $iter) use ($predicate): iterable {
+                            foreach ($iter as $key => $value) {
+                                yield $key => $predicate($value);
+                            }
+                        };
+                    }
+
+                    $res = map(function(int $i): string { return (string) $i; })([1,2,3]);
+                ',
+                'assertions' => [
+                    '$res' => 'iterable<mixed, string>',
+                ],
+            ],
+            'infersCallableReturnTypes' => [
+                '<?php
+                    /**
+                     * @template T
+                     * @template U
+                     * @param callable(T): U $predicate
+                     * @return callable(iterable<T>): iterable<U>
+                     */
+                    function map(callable $predicate): callable {
+                        return function($iter) use ($predicate): iterable {
+                            foreach ($iter as $key => $value) {
+                                yield $key => $predicate($value);
+                            }
+                        };
+                    }
+
+                    $res = map(function(int $i): string { return (string) $i; })([1,2,3]);
+                ',
+                'assertions' => [
+                    '$res' => 'iterable<mixed, string>',
+                ],
+            ],
+            'infersReturnTypeFromReturnedCallable' => [
+                '<?php
+                    /**
+                     * @template T
+                     * @template U
+                     * @param callable(T): U $predicate
+                     */
+                    function map(callable $predicate): callable {
+                        return
+                        /**
+                         * @param iterable<T> $iter
+                         * @return iterable<U>
+                         */
+                        function($iter) use ($predicate): iterable {
+                            foreach ($iter as $key => $value) {
+                                yield $key => $predicate($value);
+                            }
+                        };
+                    }
+
+                    $res = map(function(int $i): string { return (string) $i; })([1,2,3]);
+                ',
+                'assertions' => [
+                    '$res' => 'iterable<mixed, string>',
+                ],
+            ],
+//            'infersNestedCallableReturnTypes' => [
```

Comment:
You can use `SKIPPED-` instead of commenting this out

Reply:
Ah, neat, will do.